src/Adapter/Product/AttributeGroup/QueryHandler/GetProductAttributeGroupsHandler.php

```diff
@@ -65,7 +65,11 @@ public function handle(GetProductAttributeGroups $query): array
             return [];
         }
 
-        $attributeGroupEntities = $this->attributeGroupRepository->listOrderedAttributeGroups($query->withAttributes(), $attributeIds);
+        $attributeGroupEntities = $this->attributeGroupRepository->listOrderedAttributeGroups(
+            $query->withAttributes(),
+            $query->getShopConstraint(),
+            $attributeIds
```

Comment:
To be consistent, shouldn't the `$attributeIds = $this->attributeRepository->getProductAttributesIds($query->getProductId());` also be based on the shop constraint?